saviSimulation/savi/jason_processing/UxVBehavior.java

```diff
@@ -73,43 +73,22 @@ public double getCompassAngle() {
 	 * update
 	 * process actions from the queue, update the UAS state variable and set the new perceptions
 	 */
-	public void update(double simTime, int perceptionDistance, List<WorldObject> objects){
-		//Process actions to update speedVal & compassAngle
-		processAgentActions();
-		
-		//Update simTime
-		double timeElapsed =  simTime - this.time; //elapsed time since last update 
-		this.time = simTime;
-		
-		//Calculate new position
-		double cosv = Math.cos(this.compasAngle);
-		double sinv = Math.sin(this.compasAngle);
-		this.position.add(new PVector(Math.round(cosv*this.speedVal*timeElapsed), Math.round(sinv*this.speedVal*timeElapsed), 0));
-		
-		//Calculate visible items
-		this.visibleItems = new ArrayList<CameraPerception>();
-		
-		//Calculate objects detected with camera	
-		for (CameraPerception c: objectDetection(objects, perceptionDistance)) {
-			visibleItems.add(c);
-		}	
 
-		//Communicate through Wifi
-		//wifiCommunication(WIFI_PERCEPTION_DISTANCE, objects);
-			
-		//Update percepts	
-		updatePercepts();
-		//this.notifyAgent(); //this interrupts the Jason if it was sleeping while waiting for a new percept.
+	//To be override on derived classes
+	public void update(double simTime, int perceptionDistance, List<WorldObject> objects){
 	}
```

Comment:
make class abstract if it makes sense

Reply:
I made the UxV abstract class, but making UxVBehavior abstract break everything.

Reply:
I don't think it makes that much sense as it is designed to UxV having a UxVBehavior instance inside. This way every UgV and UaV when declare its behavior are using that instance. This simplifies the code.